source/agora/test/QuorumShuffle.d

```diff
@@ -54,8 +54,9 @@ unittest
     void checkDistance (Height height)
     {
         auto enrolled_height = Height(GenesisValidatorCycle * ((height.value - 1) / GenesisValidatorCycle));
-        auto required_distance = (height - enrolled_height - 1) % GenesisValidatorCycle;
-        log.trace("check distance is {} for generating block at height {} with enrollments at height {}", required_distance, height, enrolled_height);
+        auto required_distance = (height - enrolled_height) % GenesisValidatorCycle;
```

Comment:
Is there any reason why the `-1` is omitted from the formula, `(height - enrolled_height - 1)`?

Reply:
Yes it seems to fix it.

Reply:
The formula should be restored, I think. According to your current code, if the `height` is 20 and the `enrolled_height` is 0, the `required_distance` is 0. The `required_distance` should be 19.

Reply:
Yes I also just noticed that locally. Perhaps we have actually uncovered a bug in the implementation or somewhere else in the test code. Will look into it. At least the test is more stable for other PRs again.